### ISSUE TITLE
Psr18Client requires allow_url_fopen

### ISSUE THREAD

**Issue by araab** [symfony/symfony#41798]
**Symfony version(s) affected**: 5.3.2

**Description**
Psr18Client from http-client requires allow_url_fopen to be active.
I implemented a test command with the example from the official documentation:
https://symfony.com/doc/current/http_client.html#psr-18-and-psr-17

With allow_url_fopen it works, but when it is disabled I get the following error:

```
$ php -d allow_url_fopen=off bin/console app:test
[critical] Error thrown while running command "app:test". Message: "Warning: fopen(): symfony:// wrapper is disabled in the server configuration by allow_url_fopen=0"


In StreamWrapper.php line 74:


  Warning: fopen(): symfony:// wrapper is disabled in the server configuration by allow_url_fopen=0



app:test [--option1] [--] [<arg1>] 
```

**How to reproduce**
```
    protected function execute(InputInterface $input, OutputInterface $output): int
    {
        $r = $this->request->createRequest('GET', 'https://symfony.com/versions.json');
        $response = $this->client->sendRequest($r);
        var_dump($response->getBody()->getContents());


        return Command::SUCCESS;
    } 
```

**Comment by stof** [comment]
Well, the NativeHttpClient indeed needs allow_url_fopen. The CurlClient does not (but it requires the curl extension).

If you forbid opening url with PHP stream functions and you don't have the curl extension, you have no way to make HTTP requests here.

**Comment by derrabus** [comment]
I agree. There isn't much we can do here.

**Comment by stof** [comment]
maybe we can improve the DX by addign a check for `allow_url_fopen` in `HttpClient::create` before creating a NativeHttpClient

**Comment by araab** [comment]
but the curl extension is installed and the CurlClient is working. In https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/HttpClient/Psr18Client.php#L60 a CurlClient is created. Isn't the problem in https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/HttpClient/Response/StreamWrapper.php#L74 which returns a symfony:// url?

**Comment by stof** [comment]
@araab can you run your command in verbose mode so that we get the stack trace of the error ?

**Comment by araab** [comment]
sure. thanks for your help.

```
[info] Request: "GET https://symfony.com/versions.json"

[info] Response: "200 https://symfony.com/versions.json"

[critical] Error thrown while running command "app:test -vvv". Message: "Warning: fopen(): symfony:// wrapper is disabled in the server configuration by allow_url_fopen=0"

[debug] Command "app:test -vvv" exited with code "1"


In StreamWrapper.php line 74:
                                                                                                     
  [ErrorException]                                                                                   
  Warning: fopen(): symfony:// wrapper is disabled in the server configuration by allow_url_fopen=0  
                                                                                                     

Exception trace:
  at /tmp/my_project_name/vendor/symfony/http-client/Response/StreamWrapper.php:74
 Symfony\Component\HttpClient\Response\StreamWrapper::createResource() at /tmp/my_project_name/vendor/symfony/http-client/Response/CommonResponseTrait.php:123
 Symfony\Component\HttpClient\Response\CurlResponse->toStream() at /tmp/my_project_name/vendor/symfony/http-client/Psr18Client.php:108
 Symfony\Component\HttpClient\Psr18Client->sendRequest() at /tmp/my_project_name/src/Command/TestCommand.php:43
 App\Command\TestCommand->execute() at /tmp/my_project_name/vendor/symfony/console/Command/Command.php:299
 Symfony\Component\Console\Command\Command->run() at /tmp/my_project_name/vendor/symfony/console/Application.php:996
 Symfony\Component\Console\Application->doRunCommand() at /tmp/my_project_name/vendor/symfony/framework-bundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /tmp/my_project_name/vendor/symfony/console/Application.php:295
 Symfony\Component\Console\Application->doRun() at /tmp/my_project_name/vendor/symfony/framework-bundle/Console/Application.php:82
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /tmp/my_project_name/vendor/symfony/console/Application.php:167
 Symfony\Component\Console\Application->run() at /tmp/my_project_name/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:56
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at /tmp/my_project_name/vendor/autoload_runtime.php:35
 require_once() at /tmp/my_project_name/bin/console:11

app:test [--option1] [--] [<arg1>]
```

**Comment by derrabus** [comment]
Your stack trace tells us that you indeed use the cURL client.

Maybe @nicolas-grekas has some insights here: Would it be very difficult to change the cURL client to not rely on custom steam wrappers?

**Comment by nicolas-grekas** [comment]
Thanks for the details, this is fixed in #41807

**Comment by araab** [comment]
thank you for the fast fix
